calibration_assistant.cpp

```diff
@@ -0,0 +1,1250 @@
+        m_pCurrDec->SetValue(wxString::Format("%+.1f", dec));
+        if (!m_meridianFlipping)
+        {
+            if (fromTimer)
+            {
+                if (fabs(lastDec - dec) > 10)
+                {
+                    if (lastDec != INVALID_DEC)
+                        ShowExplanationMsg(dec);
+                    lastDec = dec;
+                }
+            }
+            else
+                ShowExplanationMsg(dec);
+        }
+    }
+    else
+    {
+        ShowError(_("Mount can't report its pointing position"), true);
+        if (m_pTimer)
+            m_pTimer->Stop();
+    }
+}
+// Return of 'true' means error occurred
+bool CalibrationAssistant::GetCalibPositionRecommendations(int* HA, int* Dec) const
+{
+    double hourAngle;
+    int bestDec;
+    int bestOffset;
+    double lst;
+    double ra;
+    double dec;
+    double eqAltitude;
+    bool pointingEast = false;
+    bool errorSeen = false;
+
+    bestDec = defBestDec;
+    bestOffset = defBestOffset;
+    try
+    {
+        if (pPointingSource && pPointingSource->CanReportPosition())
+        {
+            if (pPointingSource->PreparePositionInteractive())
+                return true;
+            if (pPointingSource->GetCoordinates(&ra, &dec, &lst))
+            {
+                throw ERROR_INFO("CalPositionRecommendations: Mount not reporting pointing position");
+            }
+            hourAngle = norm(lst - ra, -12.0, 12.0);
+            if (hourAngle <= 0)
+                bestOffset = -bestOffset;
+            // Check that we aren't pointing down in the weeds at the default Dec location
+            double lat, lon;
+            if (pPointingSource->GetSiteLatLong(&lat, &lon))
+                bestDec = 0;
+            else if (lat >= 0)
+            {
+                eqAltitude = 90.0 - lat;
+                if (eqAltitude < 30)        // far north location
+                    bestDec += 30 - eqAltitude;
+            }
+            else
+            {
+                eqAltitude = 90 + lat;
+                if (eqAltitude < 30)        // far south location
+                    bestDec -= (30 - eqAltitude);
+            }
+            *HA = bestOffset;
+            *Dec = bestDec;
+            m_pSlewBtn->Enable(pPointingSource->CanSlew());
+        }
+        else
+        {
+            throw ERROR_INFO("CalPositionRecommendations: mount not connected or not reporting pointing info");
+        }
+    }
+    catch (const wxString& msg)
+    {
+        POSSIBLY_UNUSED(msg);
+        *HA = defBestOffset;
+        *Dec = defBestDec;
+        errorSeen = true;
+    }
+    return errorSeen;
+}
+
+void CalibrationAssistant::ShowExplanationMsg(double dec)
+{
+    wxString slewCond;
+    if (pPointingSource->CanSlew())
+        slewCond = _("Use the 'slew' button to move the scope as close as possible to Dec = 0. ");
+    else
+        slewCond = _("Slew the scope as close as possible to Dec = 0. ");

```

Comment:
should we maybe tone down "as close as possible" -- I'm imagining people obsessively trying to zero-out their dec unnecessarily.
Maybe something like: "Use the Slew button to move the scope to within 15 degrees or so of Dec = 0"

Reply:
This is funny.  I intended it to mean "as close as practical for your site" but as you say, we would have people spending 20 minutes trying to completely zero out the Dec.  We know that many people have significant site visibility problems, some quite severe.  Hopefully the changes will work for both cases.